service/dap/server_test.go

```diff
@@ -5026,7 +5026,16 @@ func runNoDebugSession(t *testing.T, client *daptest.Client, launchRequest func(
 	client.ExpectOutputEventProcessExited(t, exitStatus)
 	client.ExpectTerminatedEvent(t)
 	client.DisconnectRequestWithKillOption(true)
-	client.ExpectDisconnectResponse(t)
+	m := client.ExpectMessage(t)
+	switch m.(type) {
+	case *dap.DisconnectResponse:
+	case *dap.OutputEvent:
+		// There may be a race between closing the exited channel, and trying
+		// to kill the process, so we may get a "Terminating process" output
```

Comment:
Kill-after-undetected-exit doesn't hurt, does it? So the only problem is inconsistent messages?
What about closing the channel before issuing process existed message and the terminated event? Then this case won't flake. And the case where the user clicks disconnect after process exists, but before everything is reflected is very unlikely.

Reply:
Done.